src/snowflake/snowpark/mock/_functions.py

```diff
@@ -336,6 +337,10 @@ def mock_to_date(
 
         [x] For all other values, a conversion error is generated.
     """
+
+    if isinstance(column.sf_type.datatype, DateType):
```

Comment:
When a column of type x gets casted to type x, it shouldn't raise error. Same for the changes below